From 21c2b58d118b4d4712a0257945beeae2f345b8b5 Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Thu, 2 Oct 2025 16:17:05 +0200
Subject: [PATCH 405/553] Mark the remaining cbor/json headers as
 security-significant

They only contain declarations; qcborstream.h only includes other
headers without interfering with their contents, so even though the
components are security-critical, these headers get the
header-decls-only get-out-of-jail-free card.

Amends cd4813a3c304a6d9d5d8096d91a8fc297818d624.

QUIP: 23
Task-number: QTBUG-135194
Pick-to: 6.8
Change-Id: I67bf7f8c1fdc5e6881df37207236eb3c294535ab
Reviewed-by: Edward Welbourne <edward.welbourne@qt.io>
(cherry picked from commit 6bb3171e312d1caf3c29c00bcd2cc366b98a5bd1)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/serialization/qcborstream.h   | 1 +
 src/corelib/serialization/qjsonparser_p.h | 1 +
 src/corelib/serialization/qjsonwriter_p.h | 1 +
 3 files changed, 3 insertions(+)

diff --git a/src/corelib/serialization/qcborstream.h b/src/corelib/serialization/qcborstream.h
index 7850d266639..e51296f3cca 100644
--- a/src/corelib/serialization/qcborstream.h
+++ b/src/corelib/serialization/qcborstream.h
@@ -1,6 +1,7 @@
 // Copyright (C) 2019 The Qt Company Ltd.
 // Copyright (C) 2018 Intel Corporation.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:header-decls-only
 
 #ifndef QCBORSTREAM_H
 #define QCBORSTREAM_H
diff --git a/src/corelib/serialization/qjsonparser_p.h b/src/corelib/serialization/qjsonparser_p.h
index 6b70af38152..8951bb3129b 100644
--- a/src/corelib/serialization/qjsonparser_p.h
+++ b/src/corelib/serialization/qjsonparser_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:header-decls-only
 
 #ifndef QJSONPARSER_P_H
 #define QJSONPARSER_P_H
diff --git a/src/corelib/serialization/qjsonwriter_p.h b/src/corelib/serialization/qjsonwriter_p.h
index 446ed906e21..10b89f3c106 100644
--- a/src/corelib/serialization/qjsonwriter_p.h
+++ b/src/corelib/serialization/qjsonwriter_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:header-decls-only
 
 #ifndef QJSONWRITER_P_H
 #define QJSONWRITER_P_H
-- 
2.51.2

